src/argus_ticket_rt.py

```diff
@@ -46,14 +43,37 @@ def import_settings(cls):
 
         return endpoint, authentication, ticket_information
 
+    @staticmethod
+    def convert_tags_to_dict(tag_dict: dict) -> dict:
+        incident_tags_list = [entry["tag"].split("=") for entry in tag_dict]
+        return {key: value for key, value in incident_tags_list}
+
+    @staticmethod
+    def get_custom_fields(ticket_information: dict, serialized_incident: dict) -> dict:
+        incident_tags = RequestTrackerPlugin.convert_tags_to_dict(
+            serialized_incident["tags"]
+        )
+        custom_fields = ticket_information.get("custom_fields", {})
+        custom_fields_mapping = ticket_information.get("custom_fields_mapping", {})
+
+        for key, field in custom_fields_mapping.items():
+            if type(field) is dict:
+                # Information can be found in tags
+                custom_fields[key] = incident_tags[field["tag"]]
+            else:
+                if serialized_incident[field] == "infinity":
+                    continue
+                custom_fields[key] = serialized_incident[field]
+
+        return custom_fields
+
     @staticmethod
     def create_client(endpoint, authentication):
         """Creates and returns a RT client"""
         try:
-            client = Client(
-                username=authentication["username"],
-                password=authentication["password"],
-                endpoint=endpoint,
+            client = Rt(
+                url=urljoin(endpoint, "REST/2.0"),
+                token=authentication["token"],
```

Comment:
If the library supports both tokens and username/pasword, why are you limiting it to just token?